### DISCUSSION TITLE
Como pegar a posição de um elemento com AngularJS sem utilizar classes ou ids

### DISCUSSION THREAD

**Discussion by woliveiras** [frontendbr/forum#477] [Perguntas e Respostas]
Galera, alguém conhece uma maneira de pegar onde está um elemento no HTML quando o template de uma diretiva é renderizada pelo AngularJS?

**Explicando meu problema atual:**

Eu tenho uma diretiva de popover e um botão que faz essa diretiva aparecer mudando uma variável para true. Ela está oculta pelo ng-if.

O que acontece é que eu preciso saber onde esse elemento (a div do popover) está na tela para posicionar ele exatamente em cima do botão que fez ele aparecer, porém antes do ng-if receber true esse elemento não existe na tela.

O que não consegui encontrar foi uma maneira de quando o ng-if fizer o elemento aparecer na tela eu pegar onde ele está.

Preciso fazer isso sem usar classe ou id, pois pode existir mais que um popover aparecendo na tela ao mesmo tempo e eu preciso pegar somente o que for renderizado quando eu clicar no botão.

**Comment by fabiomirandaa** [comment]
Se você criar uma classe ou id dinâmico, controlado por alguma variável na controller, ficaria ruim?

**Comment by woliveiras** [comment]
Eu ainda não tinha pensado nisso, @fabiomirandaa. Vou analisar essa opção! :D

**Comment by fabiomirandaa** [comment]
Conseguiu resolver o problema @woliveiras ?

**Comment by woliveiras** [comment]
Fala @fabiomirandaa 

Eu acabei esquecendo de dar um retorno. 

Eu tinha que fazer a busca pelo elemento logo depois do ng-if, porém não tem como fazer isso sem dar uma alterada no funcionamento do Angular ou fazer uma gambiarrinha (não que eu tenha achado).

Então eu decidi usar classes mesmo com o ng-class e colocar uma classe tipo: top, bottom, right, left.
